src/extension/mapml/src/main/java/org/geoserver/web/demo/MapMLFormatLink.java

```diff
@@ -32,9 +42,50 @@ public ExternalLink getFormatLink(PreviewLayer layer) {
 
     /** Customize the request to use the MapML format and a native MapML CRS if possible */
     void customizeRequest(GetMapRequest request, Map<String, String> params) {
+        WMS wms = WMS.get();
+        PublishedInfo layerInfo;
+        MetadataMap metadata;
+        layerInfo = wms.getLayerByName(params.get(LAYERINFO_LAYERS));
+        if (layerInfo == null) {
+            layerInfo = wms.getLayerGroupByName(params.get(LAYERINFO_LAYERS));
+            metadata = layerInfo.getMetadata();
+        } else {
+            metadata = ((LayerInfo) layerInfo).getResource().getMetadata();
+        }
+        boolean useFeatures = Boolean.parseBoolean(
+                metadata.get(MapMLConstants.MAPML_USE_FEATURES) != null
+                        ? metadata.get(MapMLConstants.MAPML_USE_FEATURES).toString()
+                        : FORMAT_OPTION_DEFAULT);
```

Comment:
Block repeated 3 times with just the constant being different

Reply:
Done